packages/parameters/src/dynamodb/DynamoDBProvider.ts

```diff
@@ -3,6 +3,7 @@ import {
   DynamoDBClient,
   GetItemCommand,
   paginateQuery,
+  type DynamoDBPaginationConfiguration,
```

Comment:
This probably slipped unintentional

Reply:
No, it's intentional. 

In a recent update of the AWS SDK for DynamoDB it appears that the type/shape of the method to paginate (that we [use here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/parameters/src/dynamodb/DynamoDBProvider.ts#L416)) has changed. Based on the error that TS was giving me I was able to track down the change to [this PR](https://github.com/aws/aws-sdk-js-v3/pull/5288) that was part of v[3.490.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.490.0). 

The PR appears to have added new properties to the DynamoDB client type. On the other hand we were using a more generic type coming from `@aws-sdk/types`. In order to fix the issue I removed the import from `@aws-sdk/types` and used instead the more specific (and correct type).

My bad for not documenting the change more explicitly in the PR.